flink-kubernetes/src/test/java/org/apache/flink/kubernetes/KubernetesResourceManagerTest.java

```diff
@@ -398,4 +421,19 @@ private void terminatePod(Pod pod) {
 				.build())
 			.build());
 	}
+
+	private FlinkKubeClient createFlinkKubeClient() {
+		ExecutorService kubeClientExecutorService = Executors.newDirectExecutorService();
+		return new Fabric8FlinkKubeClient(flinkConfig, kubeClient, () -> kubeClientExecutorService) {
+			@Override
+			public CompletableFuture<Void> createTaskManagerPod(KubernetesPod kubernetesPod) {
+				return CompletableFuture.runAsync(
+					() -> {
+						throw new RuntimeException("Exception");
+					},
+					kubeClientExecutorService
+				);
```

Comment:
```suggestion
				return CompletableFuture.completeExceptionally(new RuntimeException("Exception"));
```
This may be good enough for the test.

Reply:
I am not what do you mean here. Since the `createTaskManagerPod` needs to return a `CompletableFuture<Void>`.

Reply:
Sorry, I meant you can return `FutureUtils.completedExceptionally(new RuntimeException("Exception"))`. This will create already completed with an Exception future.

Reply:
Nice suggestion.